packages/wallet/src/TxBuilder/types.ts

```diff
@@ -0,0 +1,126 @@
+import { Cardano } from '@cardano-sdk/core';
+import { CustomError } from 'ts-custom-error';
+import { DeepPartial } from '@cardano-sdk/util';
+import { InputSelectionError } from '@cardano-sdk/cip2';
+
+import { OutputValidation } from '../types';
+import { SignTransactionOptions } from '../KeyManagement';
+
+// Does not transform TokenMap into Partial<TokenMap> because it would also make it's methods and map types optional
+// and that forces users to do many checks before being able to use any of it's properties and methods
+export type PartialTxOut = DeepPartial<Cardano.TxOut, Cardano.TokenMap>;
```

Comment:
This type doesn't look right
![image](https://user-images.githubusercontent.com/2457868/187845319-a5282bba-5563-47ea-b9d7-d5877096e5e9.png)

Reply:
It's because DeepPartial dived into Cardano.Address type and made all it's properties optional. It's not necessarily bad, but I can add it to the Omit types

Reply:
Maybe something like: `Partial<Pick<TxOut, 'address' | 'datum'>> & {value: Partial<Value>}`

Reply:
Yeap, this works fine. I'll update it

Reply:
Fixed in [d092381](https://github.com/input-output-hk/cardano-js-sdk/pull/396/commits/d092381a0921bb8258d6ebe3a1baca6f50a0f540)